matching/pom.xml

```diff
@@ -133,4 +114,96 @@
       </plugin>
     </plugins>
   </build>
+
+  <profiles>
+    <profile>
+      <id>linux64</id>
+      <activation>
+        <os>
+          <family>Unix</family>
+          <name>!mac os x</name>
+          <arch>amd64</arch>
+        </os>
+      </activation>
+
+      <dependencies>
+        <dependency>
+          <groupId>org.kframework.mpfr_java</groupId>
+          <artifactId>mpfr_java</artifactId>
+          <version>1.3</version>
+          <classifier>linux64</classifier>
+        </dependency>
+      </dependencies>
+
+      <properties>
```

Comment:
I'm pretty sure you don't need these because it should inherit them from the parent pom.

Reply:
The "host OS only change" is in response to a problem with the way `hawtjni` packages the native code. It allows you to attach whatever classifier you like to the JAR filename, but requires that the _internal_ structure of the JAR uses its naming structure.

This produces JARs that look something like:
```
osx-arm.jar
|-- META-INF
  |-- osx64
    |-- native
      |-- lib.dylib [arm library]

osx-x86.jar
|-- META-INF
  |-- osx64
    |-- native
      |-- lib.dylib [x86 library]
```

When the two dependencies are _both_ specified, the two copies of `lib.dylib` collide, and we end up potentially selecting the wrong version. If you depend on only one at a time, the problem goes away.

The options I considered to fix the problem were:
1. Modify the `hawtjni` packaging code (impractical, more trouble than it's worth)
2. Modify _our_ packaging code (i.e. the copying of native libraries from JARs. I suspect this might also be a hassle).
3. Select only one native dependency at Maven time. This also appears to be an issue, as you identify.

Re. the parent POM, my understanding is that `matching` gets built even if the backend isn't part of the K distribution repo. The dependency was there beforehand, and I've just reconfigured it.

Let me know what you reckon the best way to resolve these issues is.